cmd/operator-sdk/new/cmd.go

```diff
@@ -100,8 +100,13 @@ generates a default directory layout based on the input <project-name>.
 	newCmd.Flags().StringVar(&repo, "repo", "",
 		"Project repository path for Go operators. Used as the project's Go import path. This must be set if "+
 			"outside of $GOPATH/src (e.g. github.com/example-inc/my-operator)")
+
+	// todo(camilamacedo86): remove before 1.0.0
 	newCmd.Flags().BoolVar(&gitInit, "git-init", false,
 		"Initialize the project directory as a git repository (default false)")
+	_ = newCmd.Flags().MarkDeprecated("git-init", "This flag will be removed soon. "+
```

Comment:
Check this error with a panic, since this should never occur.

Reply:
Should we really give a panic if the flag was not marked as deprecated? 
IMO shows fine in this case just suppress the error.

Reply:
Right now the flag output is
```
Flag --git-init has been deprecated, This flag will be removed soon. To continue using git, run `git init` once your project is created.
```
The following reads better:
```
Flag --git-init has been deprecated, instead run `git init` once your project is created to use git
```

So change the help text to:
```suggestion
	_ = newCmd.Flags().MarkDeprecated("git-init", "instead run `git init` once your project is created to use git")
```

Reply:
@camilamacedo86 `MarkDeprecated` will only error if the flag name is not found, which it always will be. If it isn't the SDK should panic, like in other commands.

Reply:
`which it always will be` . Exactly, because of this shows not a scenario that really needs to be covered. However, it is fine. The suggestion was applied.